src/app/shared/components/display-table/display-table.component.ts

```diff
@@ -1,25 +1,10 @@
 import { Component, Input } from '@angular/core';
-import { HelperService } from '../../services/helper.service';
 
 @Component({
   selector: 'app-display-table',
   templateUrl: './display-table.component.html',
   styleUrls: ['./display-table.component.css'],
 })
 export class DisplayTableComponent {
-  @Input()
-  get report() {
-    return this._report;
-  }
-  set report(report: any) {
-    this._report = report;
-  }
-
-  public _report: any = {};
-
-  constructor(private helperService: HelperService) {}
-
-  getCheckpointType(type: number) {
-    return this.helperService.getCheckpointType(type);
-  }
+  @Input() report!: any;
```

Comment:
You have enough report types to choose from right? Or is it not that straight forward once again?